src/board_controller/build.cmake

```diff
@@ -140,17 +144,34 @@ target_include_directories (
     ${CMAKE_CURRENT_SOURCE_DIR}/src/board_controller/emotibit/inc
     ${CMAKE_CURRENT_SOURCE_DIR}/src/board_controller/ntl/inc
     ${CMAKE_CURRENT_SOURCE_DIR}/src/board_controller/aavaa/inc
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/board_controller/pieeg/inc
 )
 
 target_compile_definitions(${BOARD_CONTROLLER_NAME} PRIVATE NOMINMAX BRAINFLOW_VERSION=${BRAINFLOW_VERSION})
 
 set_target_properties (${BOARD_CONTROLLER_NAME}
-    PROPERTIES
-    ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/compiled
-    LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/compiled
-    RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/compiled
+PROPERTIES
+ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/compiled
+LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/compiled
+RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/compiled
 )
 
+find_package(Threads REQUIRED)
+
+if (ANDROID)
+    target_link_libraries(${BOARD_CONTROLLER_NAME} PRIVATE log android c m dl)
+else()
+    target_link_libraries(${BOARD_CONTROLLER_NAME} PRIVATE pthread dl)
+endif(ANDROID)
+
+target_compile_definitions(BoardController PRIVATE USE_PERIPHERY)
```

Comment:
that should be under cmake option, most of the boards do not need c_periphery

Reply:
Updated to conditionally include c_periphery.